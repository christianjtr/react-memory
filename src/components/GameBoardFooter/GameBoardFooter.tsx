import React from 'react';
import { useGameContext } from '../../hooks';

const GameBoardFooter = (): React.ReactElement => {
  const {
    state: { score, timer },
  } = useGameContext();

  return (
    <div className="flex my-5 text-xl">
      <div className="w-1/2 font-medium">{`Time: ${timer}`}</div>
      <div className="w-1/2 text-right font-medium">{`Score: ${score}`}</div>
    </div>
  );
};

export default GameBoardFooter;
