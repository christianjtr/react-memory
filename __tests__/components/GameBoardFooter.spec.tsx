import React from 'react';
import '@testing-library/jest-dom';
import { render, screen, cleanup } from '@testing-library/react';
import { createRenderer } from 'react-test-renderer/shallow';

import GameBoardFooter from '../../src/components/GameBoardFooter/GameBoardFooter';

jest.mock('../../src/hooks', () => ({
  useGameContext: jest.fn().mockReturnValue({
    state: {
      timer: 60,
      score: 100,
    },
  }),
}));

describe('GameBoardFooter component', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render the component correctly', () => {
    const renderer = createRenderer();
    const component = renderer.render(<GameBoardFooter />);

    expect(component).toBeDefined();
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('should render the correct info', () => {
    render(<GameBoardFooter />);

    const timerElement = screen.getByText('Time: 60');
    const scoreElement = screen.getByText('Score: 100');

    expect(timerElement).toBeDefined();
    expect(scoreElement).toBeDefined();
  });
});
