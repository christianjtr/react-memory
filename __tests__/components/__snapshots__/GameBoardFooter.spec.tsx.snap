// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoardFooter component should render the component correctly 1`] = `
<div
  className="md:flex w-full my-5 text-xl font-med"
>
  <div
    className="md:flex-1"
  >
    Time: 60
  </div>
  <div
    className="md:flex-1 md:text-right"
  >
    Score: 100
  </div>
</div>
`;
